code/scripting/api/objs/color.cpp

```diff
@@ -0,0 +1,73 @@
+
+#include "color.h"
+
+namespace scripting {
+namespace api {
+
+ADE_OBJ(l_Color, color, "color", "A color value");
+
+// That member syntax is pretty weird to be honest...
+static int colorVarHelper(lua_State* L, ubyte color::*member)
+{
+	color* clr = nullptr;
+	float newVal = -1.0f;
+
+	if (!ade_get_args(L, "o|f", l_Color.GetPtr(&clr), &newVal)) {
+		return ade_set_args(L, "f", -1.0f);
+	}
+
+	if (ADE_SETTING_VAR) {
+		if (newVal < 0.0f || newVal > 255.0f) {
+			LuaError(L, "Invalid color value %f!", newVal);
+			CLAMP(newVal, 0.0f, 255.0f);
+		}
+
+		clr->*member = static_cast<ubyte>(newVal);
+	}
+
+	return ade_set_args(L, "i", clr->*member);
+}
+
+ADE_VIRTVAR(Red,
```

Comment:
I suggest changing this to `RedValue` or something similar; and the same with Green and Blue.  A scripter might think that `color.Red` is a factory method that returns a Color object set to pure red.

Reply:
In Lua, the `color` "class" is not available to scripts and `color.Red` is not a valid expression so this should not be an issue.

Reply:
Well it's not my preference, but I'll approve it.